mlrun/model.py

```diff
@@ -166,6 +172,72 @@ def copy(self):
         return deepcopy(self)
 
 
+class ObjectList:
+    def __init__(self, child_class):
+        self._children = OrderedDict()
+        self._child_class = child_class
+
+    def values(self):
+        return self._children.values()
+
+    def keys(self):
+        return self._children.keys()
+
+    def items(self):
+        return self._children.items()
```

Comment:
If I understand correctly this class should act like a list ?
if so, a list doesn't expose these methods

Reply:
its a hibrid between list and dict, each name is unique